### DISCUSSION TITLE
Unable to pull public image from Github Container Registry

### DISCUSSION THREAD

**Discussion by kcak11** [community/community#27116] [Packages]
<p>I have pushed an image to Github Container Registry and made it “public”.<br>
Now when I am trying to pull it anonymously, I am getting an error:</p>
<blockquote>
<p>docker pull <a href="http://ghcr.io/ashishkumarkc/alpine:3.12" rel="noopener nofollow ugc">ghcr.io/ashishkumarkc/alpine:3.12</a></p>
</blockquote>
<blockquote>
<p>Error response from daemon: Head <a href="https://ghcr.io/v2/ashishkumarkc/alpine/manifests/3.12:" rel="noopener nofollow ugc">https://ghcr.io/v2/ashishkumarkc/alpine/manifests/3.12:</a> denied: denied</p>
</blockquote>
<p>The image is marked as “public” but still I am unable to pull it anonymously. Please help.</p>
<p><strong>Note</strong>: The <strong>docker pull</strong> command works fine after a <strong>docker login</strong>, but after a <strong>docker logout</strong> I get the above mentioned error. My assumption is that when an image is marked as “<em>public</em>” then there shouldn’t be a need to do a <em><strong>docker login</strong></em> first.</p>
<p>Am I missing something here ?</p>

**Comment by kcak11** [comment, marked as answer]
<p>This issue is now resolved. Looks like it takes some time for the “public” setting to take effect.<br>
I am now able to pull the image anonymously.</p>

**Reply by deavmi** [reply]
How did you get to pull it?

**Comment by jcansdale** [comment]
<aside class="quote no-group" data-username="kcak11" data-post="1" data-topic="180567">
<div class="title">
<div class="quote-controls"></div>
 kcak11:</div>
<blockquote>
<p>docker pull <a href="http://ghcr.io/ashishkumarkc/alpine:3.12" rel="noopener nofollow ugc">ghcr.io/ashishkumarkc/alpine:3.12 </a></p>
</blockquote>
</aside>
<p>Have you since deleted the image? I can’t see and haven’t been able to see this image.</p>

**Comment by kcak11** [comment]
<p>Yes I have deleted that image. Also as I have mentioned earlier, the issue was resolved when I tried after around 45 minutes of the image push. So looks like it was a timing issue.</p>

**Comment by jiacai2050** [comment]
If anyone still have this issue after change a private image to public, try this solution
```sh
docker logout ghcr.io
# pull again
```
- https://stackoverflow.com/a/75304123/2163429

**Reply by smyja** [reply]
What should the workflow yml file look like? because thats what i am using.
